proposals/pr002-hunkspace.md

```diff
@@ -0,0 +1,543 @@
+dimensions than the current 4 of CLDR, to allow us to address both
+resources uniquely.  Call that key space a "hunkspace".
+
+At minimum this means adding a "source" dimension to the key space,
+allowing for example "cldr.unicode.org" to be used as a (fixed)
+dimension for all CLDR data.  We could have a "fuchsia.com" source that
+could use a custom data schema like so:
+
+```
+Key: {
+  source = "fuchsia.dev",
+  app="weirdy-birds",
+  message_id=42,
+  locale="en-US"
+}
+Value: "Hello, {user}"
+```
+
+
+A note is in order on what constitutes "identical" messages.
+Two messages are identical if and only if they map to the same point in
+the hunkspace, which means that they are interchangeable no matter what
+the context.
+
+Two messages that happen to have identical strings, such as "Print" (1)
+and "Print" (2) won't necessarily be interchangeable, if one is used to
+mean "Go print this!", while the other is used to stand for "something
+which is printed".  This is because these two messages could be
+displayed differently in other languages. "Штампај", "Одштампано" would
+be the respective translations in another language. The hope here is
+that it would always be possible to split a keyspace point into two by
+providing finer-grained context where needed, but discussions of that
+are out of scope.
+
+A CLDR-only data provider would be a data provider which fixes the
+"source" string to the value "cldr.unicode.org".
+
+## Requirements
+Some requirements distilled from the discussion above:
+
+1.  There must be a commonly understood data schema for each data set.
+     For example "CLDR unicode data",
+2.  It must be possible for that data to come from multiple sources.
+     E.g. "mozilla.org", vs "google.com", vs. "unicode.org".  Every
+    source that provides a specific data schema can have own data.
+3.  The key space is multi-dimensional. Each dimension is named, and
+    the set of dimensions is specific to each data schema.  Example
+    dimension name: `message_id`.
+4.  The dimensions of the key space are [not]{.c1} ordered.
+
+# Key encoding schema
+
+Taking into account the needs for the shape and form of the keys
+discussed above, and the requirements, here are proposed mappings.
+
+It is allowed for the key encoding to have different equivalent
+schemas, which can be used where convenient.  Two proposed schemas are
+below, one for URI, another for JSON.
+
+## URI
+
+The following mapping specifies how the dimensions of the "hunkspace"
+key are mapped to an URI.
+
+* Data schema (always required):
+  * Key: `scheme`
+  * URI: "scheme"
+  * Remark: Perhaps we could sync up the namings so that we don't need to
+	maintain an active mapping like this.  Every scheme must define the data
+	schema for each value under a specific key. The data user is supposed to be
+	familiar with the data schema and able to parse the returned conforming
+	data.
+* Source (always required):
```

Comment:
Related to my previous comment: the structure of the hunk need not be tied to a specific source.  For example, you may want to request "udata://icu4x.unicode.org/dates/datesym/en_US.json".  That data could be sourced from CLDR, or it could be sourced from anyone else, as long as the returned JSON blob has the same structure.

Reply:
Agreed.  Can you cast it in an actionable form?